Azure/sap-hana
# Add output and script to copy tfstate

## Problem
There are information from deployer deployment needed for the following deployments.

## Solution
1. Add additional information into tfstate outputs section, includes:
- `deployer-id`: the random string for naming, can be used for sap_libarary
- `vnet-mgmt`: the management vnet which will be used in sap_landscape, peering
- `deployer-uai`: the user assigned identity which will be provided access to sap_libarary
2. Create local script to upload terraform.tfstate to all deployer(s).

## Test
1. cd `sap-hana/deploy/terraform/bootstrap/sap_deployer`
1. create input.json which will create two deployers:
```
{
  "infrastructure": {
    "region": "westus2",
    "resource_group": {
    }
  },
  "deployers": [{},{}],
  "sshkey": {
    "path_to_public_key": "~/.ssh/id_rsa.pub",
    "path_to_private_key": "~/.ssh/id_rsa"
  },
  "options": {
    "enable_secure_transfer": true
  }
}
```
1. terraform init
1. terraform apply -auto-approve -var-file=input.json
1. check `post_deployment.sh` generated under `default` (which is the default workspace name)
1. Run the script:
```
$ ./post_deployment.sh 
terraform.tfstate                                                                        100%   49KB 496.6KB/s   00:00    
Warning: Permanently added '52.143.110.227' (ECDSA) to the list of known hosts.
terraform.tfstate                                                                        100%   49KB 489.7KB/s   00:00
```
